### PR TITLE
Add settings for graphql batching and pg-promise idle timeout

### PR DESCRIPTION
**Pull request by Will-Howard**
This PR adds settings for two things that I think could be tweaked to improve performance:

### The batch size on BatchHttpLink
This controls how many graphql queries can be grouped together in a single request. There is some benefit to batching:
 - There is some overhead at the HTTP request level which is reduces (this includes fetching the session and user from the database)
 - You avoid the browser stalling requests (it will only allow a certain number to the same domain concurrently, around 6)
 - In principle various caches may be used more efficiently if the same server responds to several similar requests bundled together (I'm skeptical of this actually helping)

There are some costs though:
 - The HTTP request takes as long as the _slowest_ graphql query (so e.g. when you navigate to the frontpage you have to wait for recent discussion to load before the main post list will appear)
 - It puts more load on the individual server that receives the request (so there may be competition for cpu and db connections)

I'm planning to do more optimisation of this (to e.g. not group fast and slow queries together), but as a start I think reducing the max batch size may help a little with resource usage

### The idle connection timeout in pg-promise
I noticed that a lot of request traces (the majority of ones that do something non-trivial like loading the frontpage) have several `tcp.connect`s to the database:
Screenshot 2023-06-06 at 15 40 18

This is probably pg-promise increasing the number of connections in order to do more concurrent queries. The current (default) idle connection timeout is 30s, and I think the load on each individual server is intermittent enough that a lot of the connections die before the next big request comes along. I'm going to try increasing this to ~5 minutes and see if it helps

---

I've set these both to their current values by default so this won't affect LW initially, I will report back on if either of these things help though

### CODE REVIEW

File: .github/workflows/deployEABeta.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ concurrency: deploy-ea-beta
 on:
   workflow_dispatch:
   push:
-    branches: [ea-deploy]
+    branches: [wh-perf-2023-06]
```
Comment (Will-Howard) [inline .github/workflows/deployEABeta.yaml:6]: Also preemptively changing this to do some testing on the beta site

Comment (oetherington) [reply .github/workflows/deployEABeta.yaml:6]: Is it meant to be this branch, because you're missing `-settings` in the branch name if so?

Comment (Will-Howard) [reply .github/workflows/deployEABeta.yaml:6]: It's not meant to be this branch 👍

**Comment by Will-Howard** [comment]
Update on why the integration tests were failing: the pg-promise creator says (in [various](https://github.com/vitaly-t/pg-promise/issues/699) [places](https://github.com/vitaly-t/pg-promise)) that the default `idleTimeoutMillis` is 30 seconds, but it appears to actually be [10 seconds](https://github.com/brianc/node-postgres/blob/e2d8fa2dc29001e6578c4b8f111708fa96b094d0/packages/pg-pool/index.js#L97) and is set by pg-pool.

I'm surprised this caused the tests to actually fail since it didn't change the max number of connections, I would have expected it would just use the idle connections. Maybe having more available sped the tests up enough that it hit the upper limit (it appeared to be hitting the postgres limit, not the application level limit). Anyway in prod the postgres max_connections is 1713 and it hovers between 100 and 200, so I still think we can safely increase this (but not by as much)

**Review by s-cheng — APPROVED** [review]
Cool seems reasonable - your PR description was really helpful, appreciate it! :)

**Pull request merged**